dicodile/update_z/dicod.py

```diff
@@ -125,29 +124,30 @@ def dicod(X_i, D, reg, z0=None, DtD=None, n_seg='auto', strategy='greedy',
         freeze_support=freeze_support, warm_start=warm_start
     )
 
-    comm = _spawn_workers(n_workers, hostfile)
-    t_transfert, workers_segments = _send_task(comm, X_i, D, z0, DtD, w_world,
-                                               params)
+    workers = _spawn_workers(n_workers, hostfile)
+    t_transfer, workers_segments = _send_task(workers.comm, X_i,
```

Comment:
Mayeb better to just pass the `workers` and take the `comm` in `send_command_to_workers`?

Reply:
done :-)

Reply:
Much cleaner, I like it :)